src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +10,65 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
         {
             BoundExpression rewrittenReceiver = VisitExpression(node.ReceiverOpt);
 
-            // field access of a tuple actually accesses the underlying field.
-            FieldSymbol field = (node.FieldSymbol as TupleFieldSymbol)?.UnderlyingTypleFieldSymbol ??
-                                node.FieldSymbol;
+            return RewriteTupleFieldAccess(node, rewrittenReceiver);
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField == null)
+            {
+                return MakeFieldAccess(node.Syntax, rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type, node);
+            }
```

Comment:
Consider moving this case to the caller so that `RewriteTupleFieldAccess` is only called for tuple fields.